src/shared/reducers/reducer.js

```diff
@@ -60,16 +62,17 @@ const initialState = {
   },
 
   events: toByIdMap(
-    norcalMtb2016Events
-    .concat(ncnca2016Events)
-    .concat(ncnca2017Events)
-    .concat(usac2017Events)
+    norcalMtb2016Events.map(setEventsCalendarId('cal-norcal-mtb-2016'))
+    .concat(ncnca2016Events.map(setEventsCalendarId('cal-ncnca-2016')))
+    .concat(ncnca2017Events.map(setEventsCalendarId('cal-ncnca-2017')))
+    .concat(usac2017Events.map(setEventsCalendarId('cal-usac-2017')))
```

Comment:
I think this line gets confusing with this amount of manipulations. Let's refactor to:

```js
const _ = require('lodash')
//..
const setCalendarId = calendarId => event => ({...event, calendarId })
const toByIdMap = _.partialRight(_.keyBy, 'id')

const result = toByIdMap(
  _.concat(
    norcalMtb2016Events.map(setCalendarId('cal-norcal-mtb-2016')),
    ncnca2016Events.map(setCalendarId('cal-ncnca-2016')),
    //...
  )
)

```

Notice shorter name of the function `setCalendarId` and refactoring of `toByIdMap` to use lodash

---

Also, I forgot to mention that I plan to set a relationship as many-to-many. So event can belong to multiple calendars. 

Use-case would be you would want to pick events from multiple calendars and create your own or combine multiple calendars into own calendar. But every even also should point to it's "primaryCalendar" (not sure if "primary" is a good name) where it was created originally. 

With that I think we should name this property as "primaryCalendarId"  since later we would have "calendarIds"  and having singular "calendarId" along with plural "calendarIds" would be confusing. 

Open for naming ideas.

Reply:
I'd suggest using `calendarId` as it's really clear and maybe use `secondaryCalendarsIds` for others. But it's not critical.

Reply:
Refactored

Reply:
@graymur sounds good.